src/event_listeners.cc

```diff
@@ -0,0 +1,42 @@
+
+#include <iostream>
+#include <map>
+#include <string>
+
+#include "event_listeners.h"
+
+std::map<std::string, EventListener*(*)()>& get_registered_event_listeners() {
+  static std::map<std::string, EventListener*(*)()> registered_event_listeners;
+  return registered_event_listeners;
+}
+
+static std::vector<EventListener*> activated_event_listeners;

```

Comment:
```suggestion
static std::vector<std::unique_ptr<EventListener>> activated_event_listeners;

```

If you do it this way, it basically makes `cleanup_event_listeners()` irrelevant. Yay for guaranteed destructor calls.

Reply:
Thanks for the suggestion, I'll use this!